src/App.js

```diff
@@ -40,16 +40,30 @@ class Board extends Component {
         this.onChangeDescription = this.onChangeDescription.bind(this);
     }
 
-    onOpenItemDetails(itemDetails) {
-        this.setState({ itemDetails, isDisplayedDetails: true })
+    onOpenItemDetails(itemDetails, id) {
+        this.setState({ 
+            itemDetails, 
+            isDisplayedDetails: true, 
+            selectedItemId: id
+        });
     }
 
     onCloseItemDetails() {
         this.setState({...initState})
     }
 
-    onChangeDescription(description) {
-        this.setState({itemDetails: {...this.state.itemDetails, description}})
+    onChangeDescription(newDescription) {
+
+        const newItem = this.state.boardItems.filter( (item) => item.id === this.state.selectedItemId)[0] || this.state.boardItems[0];
+        newItem.params.description = newDescription;
+
+        this.setState({
+            itemDetails: newItem.params,
+            boardItems: [
+                ...this.state.boardItems.filter( (item) => item.id !== this.state.selectedItemId),
+                newItem
```

Comment:
@SammyIsra Do you see any rewiew's comment now?

Reply:
I'm using `filter` twice because first I am searching for the item that *was* edited by the user. The second time, I am searching for all items that were *not* edited by the user.